src/devices/Common/Iot/Device/Multiplexing/GpioOutputSegment.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.Gpio;
+using System.Diagnostics;
+using System.Threading;
+using Iot.Device.Multiplexing.Utility;
+
+namespace Iot.Device.Multiplexing
+{
+    /// <summary>
+    /// IOutputSegment implementation that uses GpioController.
+    /// </summary>
+    public class GpioOutputSegment : IOutputSegment, IDisposable
+    {
+        private readonly int[] _pins;
+        private readonly bool _shouldDispose;
+        private readonly CancellationToken _token;
+        private readonly VirtualOutputSegment _segment;
+        private GpioController _controller;
+
+        /// <summary>
+        /// IOutputSegment implementation that uses GpioController.
+        /// </summary>
+        /// <param name="pins">The GPIO pins that should be used and are connected.</param>
+        /// <param name="token">Cancellation token to use to notify cancelling the output segment.</param>
```

Comment:
I don't think CancellationToken should be a parameter that gets passed in to the constructor. Seems like it is really only used by the Dispay method, so I think the right way to do it would be to have the method `void Display()` instead be `Task DisplayAsync(CancellationToken ct)`